cdap-api/src/main/java/co/cask/cdap/api/dataset/InstanceNotFoundException.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.dataset;
+
+/**
+ * Thrown for an operation on a non-existing dataset instance.
+ */
+public class InstanceNotFoundException extends DatasetManagementException {
+  public InstanceNotFoundException(String message) {
```

Comment:
Good point. I copied this from existing dataset service code and tried to keep the changes minimal. But since we're at it... I'll make that change.